Automattic/woocommerce-payments
# Update connect account page

Fixes #5508 

#### Changes proposed in this Pull Request
- Remove unused experiment code.
- Implement new design.
- Migrate modified code to TypeScript.
- Refactor affected code.

Also confirmed that Jetpack connection happens just before the new onboarding flow, and every redirection to Stripe KYC lands also on the new onboarding flow, making a consistent experience.

#### Testing instructions
> **Note** All the mention **wcpaySettings** variables can be easily adjusted on [`get_js_settings`](https://github.com/Automattic/woocommerce-payments/blob/aefd3ce72e72c97b1a68f1df5baa9e8fc56dc088/includes/admin/class-wc-payments-admin.php#L704).

- Disconnect your existing account or enable `Force the plugin to act as disconnected from WCPay` in **WCPay Dev**.
- Go to **Payments**
- The connect account page should load and the design match.
- Check the following variations work as expected:
  - Clicking on `Learn more about how to receive payouts` shows a modal.
  - Setting `wcpaySettings.connect.country = ''` and clicking on `Finish setup` shows a modal.
  - Setting `wcpaySettings.errorMessage = 'Sample error message'` shows an error notice at the top.
  - Setting `wcpaySettings.onBoardingDisabled = true` only shows the disable message.

#### Screenshots
| Desktop | Mobile |
|-|-|
|<img width="690" alt="Screenshot 2023-05-12 at 08 42 46" src="https://github.com/Automattic/woocommerce-payments/assets/7670276/bf2bc2d9-331d-4550-a66a-61cc588ce0c6">|<img width="371" alt="Screenshot 2023-05-12 at 08 45 59" src="https://github.com/Automattic/woocommerce-payments/assets/7670276/6f4fd8e0-7307-4952-b764-b5ccfc8d89f0">|

| Country modal | Learn more modal |
|-|-|
|<img width="512" alt="Screenshot 2023-05-12 at 08 56 45" src="https://github.com/Automattic/woocommerce-payments/assets/7670276/d77a1f18-c724-4e7d-bcde-d7fb0f484ca2">|<img width="715" alt="Screenshot 2023-05-12 at 09 08 26" src="https://github.com/Automattic/woocommerce-payments/assets/7670276/90cdd8dc-bf46-4bdf-914c-5ac0bd8b16da">|

| Error notice | Disable message |
|-|-|
|<img width="692" alt="Screenshot 2023-05-12 at 08 57 14" src="https://github.com/Automattic/woocommerce-payments/assets/7670276/f8f6466c-f7d1-49a5-8832-f61a5accc898">|<img width="692" alt="Screenshot 2023-05-12 at 08 58 27" src="https://github.com/Automattic/woocommerce-payments/assets/7670276/2ac5db2a-4e48-4936-b69b-edb3613d7c31">|


-------------------


**Post merge**
